tests/HasTranslatableSlugTest.php

```diff
@@ -153,4 +153,83 @@ public function it_can_use_a_callback_to_generate_a_slug_per_language()
         $this->assertSame('name-en-other-en', $this->testModel->slug);
         $this->assertSame('name-nl-other-nl', $this->testModel->getTranslation('slug', 'nl'));
     }
+
+    /** @test */
+    public function it_can_handle_overwrites_when_creating_a_model()
+    {
+        $this->testModel->setTranslation('name', 'en', 'Test value EN');
+        $this->testModel->setTranslation('name', 'nl', 'Test value NL');
+        $this->testModel->setTranslation('slug', 'nl', 'updated-value-nl');
+        $this->testModel->setTranslation('slug', 'en', 'updated-value-en');
+
+        $this->testModel->save();
+
+        $this->assertSame('updated-value-en', $this->testModel->slug);
+        $this->assertSame('updated-value-nl', $this->testModel->getTranslation('slug', 'nl'));
+    }
+
+    /** @test */
+    public function it_can_handle_overwrites_when_updating_a_model()
+    {
+        $this->testModel->setTranslation('name', 'en', 'Test value EN');
+        $this->testModel->setTranslation('name', 'nl', 'Test value NL');
+        $this->testModel->save();
+
+        $this->testModel->setTranslation('slug', 'nl', 'updated-value-nl');
+        $this->testModel->setTranslation('slug', 'en', 'updated-value-en');
```

Comment:
For improved readability, update the English version before updating the nl one.